jsapp/js/components/apiTokenDisplay.tsx

```diff
@@ -0,0 +1,64 @@
+import React, {
+  useState,
+  useEffect,
+} from 'react';
+import bem from 'js/bem';
+import {dataInterface} from 'js/dataInterface';
+import TextBox from 'js/components/common/textBox';
+import Button from 'js/components/common/button';
+
+const HIDDEN_TOKEN_VALUE = '*'.repeat(40);
+
+/**
+ * Displays secret API token of a logged in user.
+ * The token is obfuscated until a "show me" button is clicked.
+ */
+export default function ApiTokenDisplay() {
+  const [token, setToken] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+  const [isVisible, setIsVisible] = useState(false);
+
+  const toggleTokenVisibility = () => {
+    setIsVisible(!isVisible);
+  };
+
+  useEffect(() => {
+    if (isVisible && token === null) {
+      const fetchToken = async () => {
+        setIsFetching(true);
+        try {
+          const result = await dataInterface.apiToken();
+          setToken(result.token);
+        } catch (error) {
+          setToken(null);
```

Comment:
This is unnecessary. token must be null already to get here.

Also, when the user leaves this page and goes back, it would be null again until clicking show. Then refetch. Right? It should be possible to sometimes update the token when reloading the page.

Reply:
Remove the unnecessary nullification 🦅 👁️